File: examples/typescript/guides/ai-agents/mock-agent.ts
/** Mock LLM and tools - no external API dependencies */

let callCount = 0;

export interface LLMResponse {
  content: string;
  toolCalls: Array<{ name: string; args: Record<string, unknown> }>;
  done: boolean;
}

export function callLlm(messages: Array<{ role: string; content: string }>): LLMResponse {
  callCount += 1;
  if (callCount === 1) {
    return {
      content: '',
      toolCalls: [{ name: 'get_weather', args: { location: 'SF' } }],
      done: false,
    };
  }
  return { content: "It's 72°F and sunny in SF.", toolCalls: [], done: true };
}

export function runTool(name: string, args: Record<string, unknown>): string {
  if (name === 'get_weather') {
    const loc = String(args?.location ?? 'unknown');
    return `Weather in ${loc}: 72°F, sunny`;
  }
  return `Unknown tool: ${name}`;
}

File: examples/go/guides/ai-agents/mock_agent.go
package main

// CallLLM is a mock - no external LLM API.
// First call returns tool_calls; second returns final answer.
var llmCallCount int

type LLMResponse struct {
	Content   string
	ToolCalls []ToolCall
	Done      bool
}

type ToolCall struct {
	Name string
	Args map[string]interface{}
}

func CallLLM(messages []map[string]interface{}) LLMResponse {
	llmCallCount++
	if llmCallCount == 1 {
		return LLMResponse{
			Content:   "",
			ToolCalls: []ToolCall{{Name: "get_weather", Args: map[string]interface{}{"location": "SF"}}},
			Done:      false,
		}
	}
	return LLMResponse{Content: "It's 72°F and sunny in SF.", ToolCalls: nil, Done: true}
}

// RunTool is a mock - returns canned results.
func RunTool(name string, args map[string]interface{}) string {
	if name == "get_weather" {
		loc := "unknown"
		if v, ok := args["location"]; ok {
			loc = v.(string)
		}
		return "Weather in " + loc + ": 72°F, sunny"
	}
	return "Unknown tool: " + name
}
